src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -2,19 +2,41 @@
 
 import de.unistuttgart.iste.gits.common.event.UserProgressLogEvent;
 import de.unistuttgart.iste.gits.generated.dto.Content;
+import de.unistuttgart.iste.gits.generated.dto.RewardChangeReason;
 import de.unistuttgart.iste.gits.reward.persistence.dao.AllRewardScoresEntity;
 import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreEntity;
+import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreLogEntry;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
+import java.time.OffsetDateTime;
 
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
 @Component
 public class FitnessScoreCalculator implements ScoreCalculator {
+    private static final double MAX_DECREASE_PER_DAY = 20;
+
     @Override
     public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores, List<Content> contents) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        UserProgressLogEvent event = UserProgressLogEvent.builder().build();
+
+        double fitnessDecrease = calculateFitnessDecrease(contents,event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
```

Comment:
This is not correct. Let's say you currently have a fitness of 100, and fitness descrease would be 20. Then this formula would return: 100 * (1 - 20) = 100 * -19 = -1900
But fitness is a value between 1 and 100. Why not just `fitnessScoreBefore.getValue() - fitnessDecrease`?